ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/UiSettingsDemoActivity.kt

```diff
@@ -0,0 +1,124 @@
+        enableMyLocation()
+        // Set all the settings of the map to match the current state of the checkboxes
+        with(map.uiSettings) {
+            isZoomControlsEnabled = isChecked(R.id.zoom_button)
+            isCompassEnabled = isChecked(R.id.compass_button)
+            isMyLocationButtonEnabled = isChecked(R.id.myloc_button)
+            isIndoorLevelPickerEnabled = isChecked(R.id.level_button)
+            isMapToolbarEnabled = isChecked(R.id.maptoolbar_button)
+            isZoomGesturesEnabled = isChecked(R.id.zoomgest_button)
+            isScrollGesturesEnabled = isChecked(R.id.scrollgest_button)
+            isTiltGesturesEnabled = isChecked(R.id.tiltgest_button)
+            isRotateGesturesEnabled = isChecked(R.id.rotategest_button)
+        }
+    }
+
+    /** On click listener for checkboxes */
+    fun onClick(view: View) {
+        if (view !is CheckBox) {
+            return
+        }
+        val isChecked: Boolean = view.isChecked
+        with(map.uiSettings) {
+            when (view.id) {
+                R.id.zoom_button -> isZoomControlsEnabled = isChecked
+                R.id.compass_button -> isCompassEnabled = isChecked
+                R.id.myloc_button -> isMyLocationButtonEnabled = isChecked
+                R.id.level_button -> isIndoorLevelPickerEnabled = isChecked
+                R.id.maptoolbar_button -> isMapToolbarEnabled = isChecked
+                R.id.zoomgest_button -> isZoomGesturesEnabled = isChecked
+                R.id.scrollgest_button -> isScrollGesturesEnabled = isChecked
+                R.id.tiltgest_button -> isTiltGesturesEnabled = isChecked
+                R.id.rotategest_button -> isRotateGesturesEnabled = isChecked
+            }
+        }
+    }
+
+    /** Returns whether the checkbox with the given id is checked */
+    private fun isChecked(id: Int) = findViewById<CheckBox>(id)?.isChecked ?: false
+
+    /** Override the onRequestPermissionResult to use EasyPermissions */
+    override fun onRequestPermissionsResult(
+            requestCode: Int,
+            permissions: Array<out String>,
+            grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this)
+    }
+
+    /**
+     * enableMyLocation() will enable the location of the map if the user has given permission
+     * for the app to access their device location.
+     * Android Studio requires an explicit check before setting map.isMyLocationEnabled to true
+     * but we are using EasyPermissions to handle it so we can suppress the "MissingPermission"
+     * check.
+     */
+    @SuppressLint("MissingPermission")
+    @AfterPermissionGranted(REQUEST_CODE_LOCATION)
+    private fun enableMyLocation() {
+        if (hasLocationPermission()) {
+            map.isMyLocationEnabled = true
+        } else {
+            EasyPermissions.requestPermissions(
```

Comment:
This function doesn't need each parameter on a new line - that is only for function declarations.

Reply:
Fixed in the latest commit